jdaviz/configs/cubeviz/helper.py

```diff
@@ -80,7 +86,11 @@ def select_wavelength(self, wavelength):
         if not isinstance(wavelength, (int, float)):
             raise TypeError("wavelength must be a float or int")
         # Retrieve the x slices from the spectrum viewer's marks
-        x_all = self.app.get_viewer('spectrum-viewer').native_marks[0].x
+        sv = self.app.get_viewer('spectrum-viewer')
+        x_all = sv.native_marks[0].x
+        if sv.state.layers[0].as_steps and GLUEJUPYTER_HAS_AS_STEPS:
```

Comment:
Do these two clauses need to be in the opposite order to avoid an error if `as_steps` isn't present?

Reply:
`as_steps` is present in the layer state as of glue 1.3.0 (and we require 1.5.0).  But before glue-jupyter 309 is released, it does nothing within bqplot (and so the length does NOT double if someone manually changes the state of `as_steps` to True).

Reply:
Got it. I'll test shortly.